File: core-examples/src/main/js/io/vertx/example/core/execblocking/exec_blocking_example.js

vertx.createHttpServer().requestHandler(function (request) {

  // Let's say we have to call a blocking API (e.g. JDBC) to execute a query for each
  // request. We can't do this directly or it will block the event loop
  // But you can do this using executeBlocking:

  vertx.executeBlocking(function (future) {

    // Do the blocking operation in here

    // Imagine this was a call to a blocking API to get the result
    try {
      Java.type("java.lang.Thread").sleep(500);
    } catch(err) {
    }

    var result = "armadillos!";

    future.complete(result);

  }, function (res, res_err) {

    if (res_err == null) {

      request.response().putHeader("content-type", "text/plain").end(res);

    } else {
      res_err.printStackTrace();
    }
  });

}).listen(8080);

File: core-examples/src/main/kotlin/io/vertx/example/core/execblocking/ExecBlockingExample.kt
import io.vertx.kotlin.common.json.*

class start : io.vertx.core.AbstractVerticle()  {
  override fun start() {

    vertx.createHttpServer().requestHandler({ request ->

      // Let's say we have to call a blocking API (e.g. JDBC) to execute a query for each
      // request. We can't do this directly or it will block the event loop
      // But you can do this using executeBlocking:

      vertx.executeBlocking({ future ->

        // Do the blocking operation in here

        // Imagine this was a call to a blocking API to get the result
        try {
          java.lang.Thread.sleep(500)
        } catch(e: Exception) {
        }

        var result = "armadillos!"

        future.complete(result)

      }, { res ->

        if (res.succeeded()) {

          request.response().putHeader("content-type", "text/plain").end(res.result())

        } else {
          res.cause().printStackTrace()
        }
      })

    }).listen(8080)

  }
}
